### ISSUE TITLE
📎 Implement `noUselessStringRaw` - `clippy/needless_raw_strings`

### ISSUE THREAD

**Issue by Conaclos** [biomejs/biome#3945]
### Description

Implement [clippy/needless_raw_strings](https://rust-lang.github.io/rust-clippy/master/#/needless_raw_strings).

**Want to contribute?** Lets you know you are interested! We will assign you to the issue to prevent several people to work on the same issue. Don't worry, we can unassign you later if you are no longer interested in the issue! Read our [contributing guide](https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md) and [analyzer contributing guide](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md).

The implementer could take some inspirations from the implementation of existing rules such as `noUselessEscapeInRegex`.

**Comment by fireairforce** [comment]
Isn't this a linter for Rust syntax? Isn't the current linter of Biome for JS?

**Comment by ematipico** [comment]
> Isn't this a linter for Rust syntax? Isn't the current linter of Biome for JS?

JavaScript has a `String.raw` function: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/raw

I presume the rule should trigger when using the function on normal strings (no template literals)

**Comment by Conaclos** [comment]
Sorry this was not clear from the description.

The rule is intended to report template literals tagged with `String.raw` without any escape sequence.

Example of code to report:

```js
String.raw`a simple ${a} string`
```

Examples of code to not report:


```js
String.raw`\n ${a}`
```

**Comment by ematipico** [comment]
@Conaclos can you provide more direction to the contributors? For example, why this rule exists, when to use it, and the end user should do to fix the code when the rule is triggered

**Comment by Conaclos** [comment]
> why this rule exists

Raw strings don't handle escape sequence like in regular strings (including untagged template literals) do.
For instance the following raw string:

```js
String.raw`\n`;
```

is equivalent to the following string:

```js
"\\n";
```

Raw strings are only useful in that regard.
Thus, if you use a raw string without any escape-like sequence, they are useless.
Because the following raw string:

```js
String.raw`a`;
```

is equva;lent to the regual rstring:

```js
"a";
```

the raw string is useless.

> when to use it

We could basically always use this rule.

> and the end user should do to fix the code when the rule is triggered

Simply turn the raw string into a template literal or even a string if there is no template placeholder:

```diff
- String.raw`a`;
+ "a";
- String.raw`a ${v}`;
+ `a ${v}`;
```

**Comment by fireairforce** [comment]
thanks for your reply, i'd like to work this~
